decompiler/structures/pseudo/instructions.py

```diff
@@ -95,11 +95,12 @@ def accept(self, visitor: DataflowObjectVisitorInterface[T]) -> T:
 class BaseAssignment(Instruction, ABC, Generic[E, F]):
     """Baseclass for Assignments an Relations."""
 
-    def __init__(self, destination: E, value: F, tags: Optional[Tuple[Tag, ...]] = None):
+    def __init__(self, destination: E, value: F, tags: Optional[Tuple[Tag, ...]] = None, writes_memory=None):
         """Initialize a new assignment operation."""
         super().__init__(tags)
         self._destination = destination
         self._value = value
+        self._writes_memory = writes_memory
```

Comment:
does not seem like a proper solution to ass this field here. You could add a case to check for GlobalVariable operands in the `writes_memory` function with less code and state as far as it seems.

Reply:
Yep, I think your suggestion could work, will try that out.

Reply:
Removed.